### PR TITLE
fix: DatePicker update unit tests

### PR DESCRIPTION
**Pull request by chrismanciero**
### Description
Update unit tests for DatePicker component

### CODE REVIEW

File: src/DatePicker/DatePicker.test.js [hunk 1]
```diff
@@ -33,27 +33,36 @@ describe('<DatePicker />', () => {
...
+        // check to make sure calendar is hidden
         expect(wrapper.state('hidden')).toBeTruthy();
 
+        // click to show calendar
         wrapper.find('input[type="text"]').simulate('click', { type: 'input' });
 
+        // check to make sure calendar is shown
         expect(wrapper.state('hidden')).toBeFalsy();
 
+        // click to hide calendar
```
Comment (greg-a-smith) [inline src/DatePicker/DatePicker.test.js]: This _seems_ like the wrong comment.  It's simulating a click in the input, which based on the next assertion, should _not_ hide the calendar.

File: src/DatePicker/DatePicker.test.js [hunk 2]
```diff
@@ -62,33 +71,57 @@ describe('<DatePicker />', () => {
...
         let arrDates = [startRangeDate, endRangeDate];
         wrapper.instance().updateDate(arrDates);
 
+        expect(wrapper.state('arrSelectedDates').length).toEqual(2);
+
+        // click on body element
+        let event = new MouseEvent('mousedown', {
+            target: document.querySelector('body')
+        });
         document.dispatchEvent(event);
```
Comment (greg-a-smith) [inline src/DatePicker/DatePicker.test.js]: I don't understand why the added `mousedown` event, but then no additional assertions.  I'm probably just missing something.

File: src/DatePicker/DatePicker.test.js [hunk 3]
```diff
@@ -101,10 +134,34 @@ describe('<DatePicker />', () => {
         expect(wrapper.state('formattedDate')).toEqual(switchFormattedDate);
         expect(wrapper.state('arrSelectedDates').length).toEqual(2);
 
+        // click on body element
+        let event = new MouseEvent('mousedown', {
+            target: document.querySelector('body')
+        });
         document.dispatchEvent(event);
```
Comment (greg-a-smith) [inline src/DatePicker/DatePicker.test.js]: I don't understand why the added `mousedown` event, but then no additional assertions.  I'm probably just missing something.

File: src/DatePicker/DatePicker.test.js [hunk 4]
```diff
@@ -175,20 +232,25 @@ describe('<DatePicker />', () => {
 
         wrapper.find('input[type="text"]').simulate('keypress', { key: 'Enter' });
 
+
+        // press Esc key
+        wrapper.find('input[type="text"]').simulate('keypress', { key: 'Esc' });
```
Comment (greg-a-smith) [inline src/DatePicker/DatePicker.test.js]: Shouldn't each `test` block end with an assertion?

File: src/DatePicker/DatePicker.test.js (hunk 1 above)
Comment (chrismanciero) [reply src/DatePicker/DatePicker.test.js]: @greg-a-smith - thanks for noticing, i just updated the comment

File: src/DatePicker/DatePicker.test.js (hunk 2 above)
Comment (chrismanciero) [reply src/DatePicker/DatePicker.test.js]: removed MouseEvent code since it was not necessary

File: src/DatePicker/DatePicker.test.js (hunk 3 above)
Comment (chrismanciero) [reply src/DatePicker/DatePicker.test.js]: added assertion to check that calendar is hidden

File: src/DatePicker/DatePicker.test.js (hunk 4 above)
Comment (chrismanciero) [reply src/DatePicker/DatePicker.test.js]: Added assertions at the end of test

File: src/DatePicker/DatePicker.test.js (hunk 2 above)
Comment (chrismanciero) [reply src/DatePicker/DatePicker.test.js]: sorry, missed that mouseDown event. Just added an assertion to check that calendar is hidden.

**Pull request merged**